doc/Data Plane Code Generation/SubClients.md

```diff
@@ -0,0 +1,178 @@
+# Customizing the generated code
+
+## SubClients
+
+There are two categories of clients: service clients and their subclients. Service clients can be instantiated and have the Client suffix. Subclients can only be created by calling factory methods on other clients (commonly on service clients) and do not have the client suffix. The service client is the entry point to the API for an Azure service.
+
+There are two approaches to organize subclients and service client.
```

Comment:
Single top-level client and client hierarchy are not mutually exclusive -- it would be reasonable for a subclient organization that reflects a client hierarchy to have a single top-level client (like `BlobServiceClient` in the blob storage client hierary.

Reply:
yes, customer can combine those two. But shall we add those into document, it make it complex and confuse customer. How about add a Note behind this two approach to indicate this combine and set BlobServiceClient as an example?